go/chat/signencrypt/codec_test.go

```diff
@@ -548,3 +563,26 @@ func TestCoverageHacks(t *testing.T) {
 		decoder.Finish()
 	})
 }
+
+func TestPrefixDifference(t *testing.T) {
+	// Test that different prefixes fail verification
+	for index, input := range plaintextInputs {
+		// Vary the chunk number, just for fun.
+		chunkNum := uint64(index)
+		sealed := sealPacket(
+			[]byte(input),
+			zeroSecretboxKey(),
+			zeroSignKey(),
+			testingPrefix(),
+			zeroChunkNonce(chunkNum))
+
+		_, err := openPacket(
+			sealed,
+			zeroSecretboxKey(),
+			zeroVerifyKey(),
+			testingPrefix()+"other",
+			zeroChunkNonce(chunkNum))
+
+		assertErrorType(t, err, BadSignature)
```

Comment:
As a sanity check / typo guard, let's make sure that the message _does_ open when we use the right prefix.